docs/src/writing_good_rules.md

```diff
@@ -28,6 +28,9 @@ julia> rrule(foo, 2)
 ==#
 ```
 
+While this is more verbose, it ensures that if an error is thrown during the `pullback` the [`gensym`](https://docs.julialang.org/en/v1/base/base/#Base.gensym) name of the local function will include the name you gave it.
+This makes it a lot simpler to debug from the stacktrace.
+
```

Comment:
This paragraph was in the `@not_implemented` section but seems to actually belong here!

Reply:
correct.
I guess text was inserted in wrong place